Gruntfile.js

```diff
@@ -0,0 +1,352 @@
+
+        var plugins = ['eloqua'];
+        for(var i = 0; i < plugins.length; i++) {
+            var type = plugins[i];
+
+            /**
+             * Output files for specific Version (when version = 1.3.0)
+             * These should never overwrite files in our s3 bucket, always should be unique.
+             * Code changes require update to version.
+             *
+             * plugins/eloqua/main-1.0.0.js
+             * plugins/eloqua/main-1.0.0.gz.js
+             */
+            grunt.file.move('dist/plugins/' + type + '/main.js', 'dist/plugins/' + type + '/' + branchFileModifier + '-' + version + '.js');
+            grunt.file.move('dist/plugins/' + type + '/main.js.gz', 'dist/plugins' + type + '/' + branchFileModifier + '-' + version + '.js.gz');
+
+            /**
+             * Output files for Major Version (when version = 1.3.0 -- majorVersion = 1)
+             * These replace the existing major release version.  The concept is that customers
+             * can use v1 and we can update it with non-breaking changes without them having to update their
+             * source.
+             *
+             * plugins/eloqua/main-v1.js
+             * plugins/eloqua/main-v1.gz.js
+             */
+            grunt.file.copy('dist/plugins/' + type + '/' + branchFileModifier + '-' + version + '.js', 'dist/plugins/' + type + '/' + branchFileModifier + '-v' + majorVersion + '.js');
+            grunt.file.copy('dist/plugins/' + type + '/' + branchFileModifier + '-' + version + '.js.gz', 'dist/plugins/' + type + '/' + branchFileModifier + '-v' + majorVersion + '.js.gz');
+        }
+    });
+
+
+    grunt.registerMultiTask('md5', 'Generates MD5 for build artifacts', function() {
+        this.files.forEach(function(file) {
+            if (typeof file.src === 'undefined') {
+                grunt.fail.warn('Files object doesn\'t exist');
+            }
+
+            var srcFiles = grunt.file.expand(file.src);
+
+            srcFiles.forEach(function(srcFile) {
+                try {
+                    var srcCode = grunt.file.read(srcFile);
+                    var md5 = require('crypto').createHash('md5').update(srcCode).digest('hex');
+                    grunt.log.writeln(srcFile+' '+md5.green);
+                } catch(err) {
+                    grunt.log.error(err);
+                    grunt.fail.warn('Fail to generate an MD5 file name');
+                }
+            });
+        });
+    });
+
+    /**
+     * Checks to make sure the AWS Credentials have been passed in.
+     * This allows us to check right away to make sure we can complete this task.
+     */
+    grunt.registerTask('awsInputCheck', 'Checks to make sure we have the correct AWS Inputs', function() {
+        var aws = grunt.config.get('aws');
+
+        var hasAllKeys = true;
+        // Loop through the array of values here and make sure they are all set.
+        awsConfigKeys.forEach(function(key) {
```

Comment:
Sure, I can do that. Next time we touch the other repo's gruntfiles, we can do that there too.